### DISCUSSION TITLE
Integration test backlog

### DISCUSSION THREAD

**Discussion by pandemicsyn** [meltano/meltano#6374] [General]
Now that we have an integration test framework in place I thought it might be good to start a discussion around what all tests we should have. I figured recent additions are probably a good jumping off point so I've compiled some test candidates based on modifications made in 2.0 and forward.

1. meltano add & remove & lock
   - add gets some use in `meltano-run`, but would be good to test some variations (and test remove and lock)
2. meltano jobs and schedules
   -  jobs gets a very basic test in `meltano-run` but we should also setup a schedule and interact some with listing for both jobs and schedules.
3. meltano state copy & move
   - we could test this as part of a "migrate from `elt` to `run`" guide
   - could probably steal from the "example integration test" [walk through](https://docs.meltano.com/contribute/tests#example-library-entry), which actually uses that scenario.
5. Lots of config/env related changes.  A dedicated "test all the config/env related things" would probably be worth it and act as one central location that we can add to and maintain.
   - https://github.com/meltano/meltano/pull/6043
   - https://github.com/meltano/meltano/pull/6278 
   - https://github.com/meltano/meltano/pull/6267
   - https://github.com/meltano/meltano/pull/6308 (landing shortly)
6. Plugin `requires:` pattern
   - Might be fine to wrap into a test as part of \# 1 above, not sure if this requires an explicit test 

/cc @tayloramurphy @aaronsteers

**Comment by edgarrmondragon** [comment]
I just added an integration for https://github.com/meltano/meltano/pull/6308
